src/librustc/session/config.rs

```diff
@@ -1304,6 +1304,9 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
         "tell the linker to strip debuginfo when building without debuginfo enabled."),
     share_generics: Option<bool> = (None, parse_opt_bool, [TRACKED],
           "make the current crate share its generic instantiations"),
+    polly: bool = (option_env!("RUSTC_FORCE_USE_POLLY").is_some(),
```

Comment:
I'd like a clarification of use of `option_env!()` here.

`option_env!()` is evaluated at compile time. This means if we set `use-polly-by-default = true` in `config.toml`, the produced `rustc` will always use Polly. Is this the intended behavior?

```console
$ echo 'use-polly-by-default = false' >> config.toml
$ ./x.py build
...
$ rustc +stage2 x.rs
##^ will compile x.rs with -Zpolly even without RUSTC_FORCE_USE_POLLY=1
```

Reply:
@DiamondLovesYou Could you put the explanation into `config.toml.example` and also above this line in `src/librustc/session/config.rs`? Thanks.

(Also, the code still [fails to link](https://github.com/rust-lang/rust/pull/50044#issuecomment-382892728).)

Reply:
@DiamondLovesYou You could also modify the bootstrap rustc (`src/bootstrap/bin/rustc.rs`) to always pass `-Z polly` when `RUSTC_FORCE_USE_POLLY` is set.